### PR TITLE
Temporary LightSpeed toolbox

### PR DESCRIPTION
**Pull request by ptomato**
https://phabricator.endlessm.com/T25262

### CODE REVIEW

**Comment by ptomato** [comment]
As an exception, I am merging this in anticipation of review in order to have it in the weekly build playthrough. Don't let that stop you from leaving review comments, I will fix them ASAP afterwards.

File: src/clippyWrapper.js [hunk 1]
```diff
@@ -18,9 +18,8 @@ const ClippyIface = `
 
 var ClippyWrapper = GObject.registerClass({
 }, class ClippyWrapper extends GObject.Object {
-    _init(props = {}) {
-        this._appId = props.appId;
-        delete props['appId'];
+    _init(appId, props = {}) {
```
Comment (manuq) [inline src/clippyWrapper.js]: :+1:

File: src/framework/customModules/card/hackableDefault.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ var HackableDefault = new Module.Class({
             ''),
     },
 
-    _init: function (props = {}) {
+    _init(props = {}) {
```
Comment (manuq) [inline src/framework/customModules/card/hackableDefault.js:76]: :+1:

File: .eslintrc.yml [hunk 3]
```diff
@@ -208,6 +208,11 @@ rules:
     - error
     - last
   space-before-blocks: error
+  space-before-function-paren:
+    - error
+    - named: never
+      anonymous: never
```
Comment (manuq) [inline .eslintrc.yml:214]: I like space in annonymous functions but this is only taste. Is good to have lint for this.

File: src/clippyWrapper.js [hunk 4]
```diff
@@ -18,10 +17,10 @@ const ClippyIface = `
 
 var ClippyWrapper = GObject.registerClass({
 }, class ClippyWrapper extends GObject.Object {
-    _init(appId, props = {}) {
+    _init(appId, props = {}, variableName = 'globalParameters') {
```
Comment (manuq) [inline src/clippyWrapper.js:20]: :+1:

File: src/LightSpeed/model.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        spawnObstacleCode: GObject.ParamSpec.string('spawnObstacleCode',
+            'Spawn obstacle code', '', _propFlags, ''),
+        updateEnemyCode: GObject.ParamSpec.string('updateEnemyCode',
+            'Update enemy code', '', _propFlags, ''),
+        setParamsCode: GObject.ParamSpec.string('setParamsCode',
+            'Set parameters code', '', _propFlags, ''),
+    },
+}, class LSModel extends ClippyWrapper {
+    _init(level, props = {}) {
+        super._init('com.endlessm.LightSpeed', props, `globalLevel${level}Parameters`);
```
Comment (manuq) [inline src/LightSpeed/model.js:35]: In the clubhouse we are assuming all JS parameters are `view.JSContext.globalParameters` so we need to change that. Glad I'm reviewing this.

File: src/LightSpeed/controlpanel.js [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+
+const VALID_SHIPS = ['spaceship', 'daemon', 'unicorn'];
+
+var LSControlPanel = GObject.registerClass({
+    GTypeName: 'LSControlPanel',
+    Template: 'resource:///com/endlessm/HackToolbox/LightSpeed/panel.ui',
+    InternalChildren: ['astronautSizeAdjustment', 'scoreTargetAdjustment',
+        'shipAccelerationAdjustment', 'shipAssetButton', 'shipSizeAdjustment',
+        'shipSpeedAdjustment', 'timeLimitAdjustment', 'variablesCodeview'],
+}, class FizzicsControlPanel extends Gtk.Grid {
```
Comment (manuq) [inline src/LightSpeed/controlpanel.js:23]: Seems you forgot to rename this class to `LightSpeedControlPanel` here?

**Review by manuq — COMMENTED** [review]
Nice toolbox!

File: src/LightSpeed/model.js (hunk 5 above)
Comment (xjuan) [reply src/LightSpeed/model.js:35]: Assumptions are bad!

**Pull request merged**